packages/utils/src/useBulkSelect/useBulkSelectWithItems.js

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useMemo } from 'react';
+import { filterSelected } from '../useTableTools/helpers';
+import { selectItemTransformer, itemIds } from './helpers';
+import useBulkSelect from './useBulkSelect';
+
+const useBulkSelectWithItems = ({ onSelect, items: propItems, filter, paginator, preselected, setPage }) => {
+  const enableBulkSelect = !!onSelect;
+  const items = propItems.map((item) => selectItemTransformer(item, preselected));
+  const total = items.length;
+
+  const filteredItems = filter ? filter(items) : items;
+  const filteredTotal = filteredItems.length;
+  const filtered = filteredTotal !== total;
+
+  const paginatedItems = paginator ? paginator(filteredItems) : filteredItems;
+  const paginatedTotal = paginatedItems.length;
+
+  const allCount = filtered ? filteredTotal : total;
+
+  const setPageMemo = useMemo(() => setPage, []);
```

Comment:
I think that this one should be `useCallback` it's basically the same, but `useMemo` is generally used for heavy computations.